conda/cli/install.py

```diff
@@ -315,82 +327,71 @@ def ensure_update_specs_exist(prefix: str, specs: list[str]):
 
 
-def install_clone(args, parser):
-    """Executes an install of a new conda environment by cloning."""
-    prefix = context.target_prefix
-    index_args = get_index_args(args)
+def _assemble_environment(
+    name: str | None = None,
+    prefix: str | None = None,
+    specs: Iterable[str] = (),
+    files: Iterable[str] = (),
+    inject_default_packages: bool = True,
+) -> Environment:
+    # First, let's create an 'Environment' for the information exposed in the CLI (no files)
+    if inject_default_packages:
+        names = {MatchSpec(pkg).name for pkg in specs}
+        for pkg in context.create_default_packages:
+            pkg_name = MatchSpec(pkg).name
+            if pkg_name not in names:
+                specs.append(pkg)
+
+    requested_packages = []
+    fetch_explicit_packages = []
 
-    # common validations for all types of installs
-    validate_install_command(prefix=prefix, command="create")
+    for spec in specs:
+        if is_package_file(spec):
+            fetch_explicit_packages.append(spec)
+        else:
+            requested_packages.append(common.arg2matchspec(spec))
+
+    # transform explicit packages into package records
+    explicit_packages = []
+    if fetch_explicit_packages:
+        if len(fetch_explicit_packages) == len(specs):
+            explicit_packages = get_package_records_from_explicit(
+                fetch_explicit_packages
+            )
+        else:
+            raise CondaValueError(
+                "cannot mix specifications with conda package filenames"
+            )
 
-    clone(
-        args.clone,
-        prefix,
-        json=context.json,
-        quiet=context.quiet,
-        index_args=index_args,
+    cli_env = Environment(
+        name=name,
+        prefix=prefix,
+        platform=context.subdir,
+        requested_packages=requested_packages,
+        explicit_packages=explicit_packages,
+        config=EnvironmentConfig.from_context(),
     )
 
+    # Now let's process potential files passed via --file
+    file_envs = []
+    for path in files:
+        # parse the file
+        spec_hook = context.plugin_manager.get_environment_specifier(path)
+        file_env = spec_hook.environment_spec(path).env
+        file_envs.append(file_env)
 
-def install(args, parser, command="install"):
-    """Logic for `conda install`, `conda update`, `conda remove`, and `conda create`."""
-    prefix = context.target_prefix
+    if context.dry_run:
+        cli_env.prefix = os.path.join(mktemp(), UNUSED_ENV_NAME)
 
-    # common validations for all types of installs
-    validate_install_command(prefix=prefix, command=command)
+    return Environment.merge(cli_env, *file_envs)
 
-    if context.use_only_tar_bz2:
-        args.repodata_fns = ("repodata.json",)
 
+def install(args, parser, command="install"):
+    """Logic for `conda install`, `conda update`, and `conda create`."""
     newenv = command == "create"
     isupdate = command == "update"
```

Comment:
Updating is a little bit different from creating an environment or installing packages into an environment. For example, it doesn't really make sense to be using environment spec plugins to be updating packages. Might need to rework this some.

Reply:
demo for an alternative https://github.com/conda/conda/compare/main...soapy1:conda:refactor-cli-layer-split-out-update?expand=1